core/store/src/flat/inlining_migration.rs

```diff
@@ -335,8 +335,16 @@ mod tests {
         ];
         populate_flat_store(&store, shard_uid, &values);
 
-        let flat_storage_manager =
-            FlatStorageManager::test(store.clone(), &vec![shard_uid], CryptoHash::default());
+        let flat_storage_manager = FlatStorageManager::new(store.clone());
+        let mut store_update = store.store_update();
+        flat_storage_manager.set_flat_storage_for_genesis(
+            &mut store_update,
+            shard_uid,
+            &CryptoHash::default(),
+            0,
```

Comment:
what value is this?

Reply:
Well, this corresponds to us removing the `FlatStorageManager::test` constructor in favor of `FlatStorageManager::new`. Check the deletion in `core/store/src/flat/manager.rs`

As to why? The test construction does some weird things like automatically set the set_flat_storage_status as ready and create flat storage for shard. This isn't cool as for tests we should ideally rely on using create_flat_storage_for_shard to explicitly and set the status to "ready" with genesis as flat head.

This is exactly what we do in production code and this would help avoid potential difference in behavior between production and tests, at the cost of being a bit more explicit about what we want to do (which is also a good thing)

Reply:
Thanks for explanation, do you think there is a value of adding comment next to 0 for readability?